docs/SILProgrammersManual.md

```diff
@@ -159,6 +159,165 @@ rules out PartialApplies is no excuse to conflate applied arguments
 with function arguments. Also, without consistent use of common
 idioms, it becomes overly burdensome to evolve these APIs over time.
 
+## `AccessedStorage` and `AccessPath`
+
+The `AccessedStorage` and `AccessPath` types formalize memory access
+in SIL. Given an address-typed SIL value, it must be possible to
+reliably identify the storage location of the accessed
+memory. `AccessedStorage` identifies an accessed storage location. It
+is a value object that requires no memory allocation and can be used
+as the hash key for that location. Extra bits are also available for
+information specific to a particular optimization pass. Its API
+provides the kind of location being accessed and information about the
+location's uniqueness or whether it is distinct from other storage.
+
+### Uniquely identified storage
+
+Two uniquely identified storage locations may only alias if their
+AccessedStorage objects are identical.
+
+`AccessedStorage::isUniquelyIdentifiedStorage()` is a weaker check that
+does not depend on exclusivity enforcement.
+
+`AccessedStorage::isUniquelyIdentifiedBase()` is a stronger check that
+depends on exclusivity enforcement.
+
+Optimization passes that remove or reorder access markers may only use
+the weaker `isUniquelyIdentifiedStorage()` check.
```

Comment:
Can you put an explicit SIL example here that shows how/why these are different? That I think would help  make this clearer.

Reply:
Interestingly, I don't *think* we need the different style unique identity checks. I wanted to merge the implementation in the PR without changing the semantics. But since then, I reviewed the exclusivity optimizations and proved to myself that they are still valid even with more aggressive checks. I'll take that risk to simplify the API and docs.